### ISSUE TITLE
Refactor carbon-components-react imports to relative imports

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon-addons-iot-react#534]
We shouldn't be importing components from `carbon-components-react` in the components. If a component in this library relies on another component in this library (Carbon proxy or not), it should be imported from a relative path, not `carbon-components-react`.

We need to have a `./src/components/ComponentName/index.js` for each component instead of importing and reexporting directly from `carbon-components-react` in `./src/index.js`.

Then, all existing imports from `carbon-componets-react` need to be refactored to use the new relative locations.

Basically looking to mirror the strategy that ibm-security has put forth in their addons repo. Their [Checkbox](https://github.com/carbon-design-system/ibm-security/blob/44a91d37eb8e1714c2ba963950cd34cba027069e/src/components/Checkbox/index.js#L6) export is a good example because it's a component that only deviates in style, not in function (js).

**Comment by sstone2423** [comment]
Hello, this looks like it would be a good first issue to get an understanding of the code base. Could I claim this issue? @tay1orjones @scottdickerson

**Comment by sstone2423** [comment]
I began editting a few of the components and have run into an odd issue. When converting the `Card` component's `SkeletonText` to use the relative path, the component fails to render with 
```
type is undefined

updateMemoComponent@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:14291:28
beginWork@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:15485:16
performUnitOfWork@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:19108:12
workLoop@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:19148:24
renderRoot@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:19231:7
performWorkOnRoot@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:20138:7
performWork@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:20050:7
performSyncWork@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:20024:3
requestWork@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:19893:5
scheduleWork@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:19707:5
scheduleRootUpdate@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:20368:3
updateContainerAtExpirationTime@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:20396:10
updateContainer@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:20453:10
ReactRoot.prototype.render@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:20749:3
legacyRenderSubtreeIntoContainer/<@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:20886:14
unbatchedUpdates@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:20255:10
legacyRenderSubtreeIntoContainer@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:20882:5
render@webpack-internal:///./node_modules/react-dom/cjs/react-dom.development.js:20951:12
render@webpack-internal:///./node_modules/@storybook/react/dist/client/preview/render.js:55:24
renderMain@webpack-internal:///./node_modules/@storybook/react/dist/client/preview/render.js:93:3
renderMain@webpack-internal:///./node_modules/@storybook/core/dist/client/preview/start.js:205:7
renderUI@webpack-internal:///./node_modules/@storybook/core/dist/client/preview/start.js:231:9
emit@webpack-internal:///./node_modules/eventemitter3/index.js:180:35
StoryStore/_this.setSelection/<@webpack-internal:///./node_modul/@storybook/client-api/dist/story_store.js:239:22
```
I have limited knowledge in the `React.memo()` function, but it seems that its blowing up because of that function.

When I attempt to import the `SkeletonText` without the memo, it renders correctly.

**Comment by tay1orjones** [comment]
@sstone2423 I agree, this is a good spot to get an understanding of the codebase. You'll touch probably every file 😄 

You found a bug with our usage of `.memo`! I did some debugging, seems we can fix by changing one line in Card.
```diff
- const OptimizedSkeletonText = React.memo(SkeletonText);
+ const OptimizedSkeletonText = React.memo(props => <SkeletonText {...props} />);
```

Feel free to deliver this in pieces over time, instead of one huge PR. We can link the PR's to the issue and keep it open until all items are finished.

**Comment by sstone2423** [comment]
Oh wow good catch! This bug was really throwing me off.

I'll go ahead and break them up every few components

**Comment by sstone2423** [comment]
Ran into a similar issue in https://github.com/IBM/carbon-addons-iot-react/blob/master/src/components/ComposedModal/ComposedModal.jsx#L60

The `styled(InlineNotification)` function is giving the same undefined error as the memo function was. When `InlineNotification` or `ModalFooter` are changed to relative import, the components are undefined and the app blows up. I've tried the same `...props` fix and double-checked the `styled-components` docs, but it looks like its being used correctly.

**Comment by sstone2423** [comment]
Tracking for completed components:
[ ] - Accordian
[ ] - AddCard
[ ] - Breadcrumb
[ ] - Button
[ ] - Card
[ ] - Checkbox
[ ] - CodeSnippet
[ ] - ComboBox
[ ] - ComposedModal
[ ] - ComposedStructured
[ ] - ContentSwitcher
[ ] - CopyButton
[ ] - Dashboard
[ ] - Datatable
[ ] - Datepicker
[ ] - Dropdown
[ ] - FileDrop
[ ] - FileUploader
[ ] - Form
[ ] - FormItem
[ ] - FormLabel
[ ] - Header
[ ] - Hero
[ ] - Icon
[ ] - ImageCard
[ ] - Inlineloading
[ ] - Link
[ ] - List
[ ] - ListBox
[ ] - Loading
[ ] - Modal
[ ] - ModalWrapper
[ ] - Multiselect
[ ] - Navbar
[ ] - notification
[ ] - Numberinput
[ ] - Orderedlist
[ ] - Overflowmenu
[ ] - Page
[ ] - PageTitlebar
[ ] - PageWizard
[ ] - Pagination
[ ] - Paginationnav
[ ] - ProgressIndicator

**Comment by sstone2423** [comment]
All PRs are open with the exception of the components that have import bugs listed above

**Comment by sstone2423** [comment]
After some thought on how best to break this issue up, I have divided the work into 2 PR's. The first PR will be adding the `index.js` files to each carbon component, and changing the import paths in the main `src/index.js`. The second PR will change each component's import paths to use relative instead of `carbon-components-react`. 

My reasoning is that if I were to split this task into more PRs, 5 components per PR for example, there would be a mismatch of how components are being imported across the library which could lead to confusion. By changing ALL components at once, per subtask, it helps to keep consistency. I'll add further descriptions in each PR

**Comment by sstone2423** [comment]
It was decided this wasn't needed

**Comment by tay1orjones** [comment]
@scottdickerson @davidicus @sstone2423 I'd like to revisit this... I was out when the related PRs came through. 
Unfortunately this has left us in a very less than ideal situation. The first PR was merged, second was not, so we have many unused `index.js` files now for each component. There is no consistency of our imports across the codebase. 

While importing from `carbon-components-react` in components and stories might make it easier to discern at a glance which components come directly from Carbon, we now have situations where stories are importing from the `index.js` and in other places vice versa importing directly from `carbon-components-react`.

This inconsistency hasn't bitten us yet, but it's very easy to have a mismatch between what's shown in storybook and what we're exporting. Worst case scenario is that bugs can very easily slip through snapshots due to a simple import error.

**Comment by kevinsperrine** [comment]
This estimate could go high depending on if we choose to continue proxying Carbon components as being discussed here: https://github.com/carbon-design-system/carbon-addons-iot-react/issues/2445.
